doc/manual/rl-next/shebang-relative.md

```diff
@@ -0,0 +1,7 @@
+synopsis: ensure nix-shell shebang uses relative path
+prs: #5088
+description: {
+
+`nix-shell` shebangs use relative paths for files, but expressions were still evaluated using the current working directory. The new behavior is that evalutations are performed relative to the script.
```

Comment:
```suggestion
`nix-shell` shebangs use the script file's location to resolve relative paths to files passed as command line arguments, but expression arguments were evaluated using the current working directory as a base path.
The new behavior is that Nix language expressions passed to `nix-shell` when used in a `#!` context are now evaluated relative to the script's file system location.
```

Reply:
This is a breaking change for example if you have a repo with a `scripts` directory and you invoke them as

```console
$ ./scripts/foo
```

in which case the `foo` script likely has a workaround for this issue,

```
#!/usr/bin/env nix-shell
#!nix-shell --expr 'import ./scripts/shell.nix { withBar = true; }'
```

 which will cause it to break when Nix is updated to a version after this PR.
 
```
error: getting status of '/home/user/src/unicorn/scripts/scripts/shell.nix': No such file or directory
```

Note the double `scripts/`.

I don't know if we should break an established command like this.
We should take this seriously and add `--relative` to opt in to the new behavior.
For what it's worth, it's also a way to "declare" that a recent Nix is needed by the script.

Reply:
@roberth here's an attempt at adding a new "--relative" flag: https://github.com/NixOS/nix/commit/2fd7ee21d11bf7b06aff674663fe618e41eff5e2